lib/std/tz.zig

```diff
@@ -0,0 +1,218 @@
+const std = @import("std.zig");
+const builtin = @import("builtin");
+
+pub const Transition = struct {
+    ts: i64,
+    timetype: *Timetype,
+};
+
+pub const Timetype = struct {
+    offset: i32,
+    flags: u8,
+    name_data: [6:0]u8,
+
+    pub fn name(self: Timetype) [:0]const u8 {
+        return std.mem.sliceTo(self.name_data[0..], 0);
+    }
+
+    pub fn isDst(self: Timetype) bool {
+        return (self.flags & 0x01) > 0;
+    }
+
+    pub fn standardTimeIndicator(self: Timetype) bool {
+        return (self.flags & 0x02) > 0;
+    }
+
+    pub fn utIndicator(self: Timetype) bool {
+        return (self.flags & 0x04) > 0;
+    }
+};
+
+pub const Leapsecond = struct {
+    occurrence: i48,
+    correction: i16,
+};
+
+pub const Tz = struct {
+    allocator: std.mem.Allocator,
+    transitions: []const Transition,
+    timetypes: []const Timetype,
+    leapseconds: []const Leapsecond,
+    footer: []const u8,
+
+    pub fn parse(allocator: std.mem.Allocator, data: []const u8) !Tz {
+        const header_size = 4 + 1 + 15 + 6 * 4;
+        if (data.len < header_size) return error.BadSize;
+
+        const magic_l = data[0..4];
+        const version_l = data[4];
+        if (!std.mem.eql(u8, magic_l, "TZif")) return error.BadHeader;
+        if (version_l != '2' and version_l != '3') return error.BadVersion;
```

Comment:
`version_l == 0` also has this format up to the gmt flag at least.

Reply:
Not sure what you are trying to say. I don't support version 0 at all and just bail as soon as possible if I see that.